.github/workflows/v2-deploy-pr.yml

```
@@ -0,0 +1,132 @@
+        with:
+          workflow_run_id: ${{ github.event.workflow_run.id }}
+          artifact_name_regex: '^pr_info$'
+          github_token: ${{ github.token }}
+
+      - name: Check if Build Context file exists
+        id: build_context
+        env:
+          context_artifact_file_name: pr_info.zip
+        run: |
+          if test -f "$context_artifact_file_name"; then
+            echo "result=$context_artifact_file_name" >> $GITHUB_OUTPUT
+          fi
+
+      - name: Unpack Build Information
+        if: steps.build_context.outputs.result != ''
+        run: unzip ${{ steps.build_context.outputs.result }}
+
+      - name: Read Build Information
+        id: read_build_info
+        if: steps.build_context.outputs.result != ''
+        uses: actions/github-script@v7
+        with:
+          script: |
+            const fs = require('fs');
+            const buildData = fs.readFileSync('pr_info.json', 'utf8');
+            return JSON.parse(buildData);
+
+      - name: Parse Pull Request Event Information
+        id: pr_info
+        if: github.event.workflow_run.event == 'pull_request' && steps.build_context.outputs.result != ''
+        run: |
+          echo '${{ steps.read_build_info.outputs.result }}' | jq -r '.id' > pr_id.txt
+          echo '${{ steps.read_build_info.outputs.result }}' | jq -r '.ref' > pr_ref.txt
+          echo '${{ steps.read_build_info.outputs.result }}' | jq -r '.ref_name' > pr_ref_name.txt
+          echo '${{ steps.read_build_info.outputs.result }}' | jq -c '.updated_demos' > updated_demos.json
+          echo '${{ steps.read_build_info.outputs.result }}' | jq -c '.deleted_demos' > deleted_demos.json
+
+          echo "pr_id=$(cat pr_id.txt)" >> $GITHUB_OUTPUT
+          echo "pr_ref=$(cat pr_ref.txt)" >> $GITHUB_OUTPUT
+          echo "pr_ref_name=$(cat pr_ref_name.txt)" >> $GITHUB_OUTPUT
+          echo "updated_demos=$(cat updated_demos.json)" >> $GITHUB_OUTPUT
+          echo "deleted_demos=$(cat deleted_demos.json)" >> $GITHUB_OUTPUT
+
+      - name: Set job outputs
+        if: github.event.workflow_run.event == 'pull_request' && steps.build_context.outputs.result != ''
+        id: set_job_outputs
+        run: |
+          echo "pr_id=${{ steps.pr_info.outputs.pr_id }}" >> $GITHUB_OUTPUT
+          echo "pr_ref=${{ steps.pr_info.outputs.pr_ref }}" >> $GITHUB_OUTPUT
+          echo "pr_ref_name=${{ steps.pr_info.outputs.pr_ref_name }}" >> $GITHUB_OUTPUT
+          echo "updated_demos=${{ steps.pr_info.outputs.updated_demos }}" >> $GITHUB_OUTPUT
+          echo "deleted_demos=${{ steps.pr_info.outputs.deleted_demos }}" >> $GITHUB_OUTPUT
+    outputs:
+      pr_id: ${{ steps.set_job_outputs.outputs.pr_id }}
+      pr_ref: ${{ steps.set_job_outputs.outputs.pr_ref }}
+      pr_ref_name: ${{ steps.set_job_outputs.outputs.pr_ref_name }}
+      updated_demos: ${{ steps.set_job_outputs.outputs.updated_demos }}
+      deleted_demos: ${{ steps.set_job_outputs.outputs.deleted_demos }}
```

I may be missing something, but the `set_job_outputs` step looks redundant. You already output all these values from the `pr_info` step:
```suggestion
    outputs:
      pr_id: ${{ steps.pr_info.outputs.pr_id }}
      pr_ref: ${{ steps.pr_info.outputs.pr_ref }}
      pr_ref_name: ${{ steps.pr_info.outputs.pr_ref_name }}
      updated_demos: ${{ steps.pr_info.outputs.updated_demos }}
      deleted_demos: ${{ steps.pr_info.outputs.deleted_demos }}
```

---

Thanks, Paul, what you're suggesting makes total sense.

I have applied these changes.